src/XCMRelayer.sol

```diff
@@ -209,4 +210,43 @@ contract AxelarXCMRelayer is Auth {
     function _parachainId() internal pure returns (bytes memory) {
         return abi.encodePacked(uint8(0), CENTRIFUGE_PARACHAIN_ID);
     }
+
+    // --- Utilities ---
+    function _centrifugeCall(bytes memory message) internal view returns (bytes memory) {
+        return abi.encodePacked(
+        // The Centrifuge liquidity-pools pallet index
+        centrifugeChainLiquidityPoolsGatewayPalletIndex,
+        // The `handle` call index within the liquidity-pools pallet
+        centrifugeChainLiquidityPoolsGatewayPalletProcessMsgCallIndex,
+        // We need to specify the length of the message in the scale-encoding format
+        messageLengthScaleEncoded(message),
+        // The connector message itself
+        message
+        );
+    }
+
+    // Obtain the Scale-encoded length of a given message. Each Liquidity Pools Message is fixed-sized and
+    // have thus a fixed scale-encoded length associated to which message variant (aka Call).
+    function messageLengthScaleEncoded(bytes memory _msg) internal pure returns (bytes memory) {
+        if (Messages.isTransfer(_msg)) {
+            return hex"8501";
+        } else if (Messages.isTransferTrancheTokens(_msg)) {
+            // A TransferTrancheTokens message is 82 bytes long which encodes to 0x4901 in Scale
+            return hex"4901";
+        } else if (Messages.isIncreaseInvestOrder(_msg)) {
+            return hex"6501";
+        } else if (Messages.isDecreaseInvestOrder(_msg)) {
+            return hex"6501";
+        } else if (Messages.isIncreaseRedeemOrder(_msg)) {
+            return hex"6501";
+        } else if (Messages.isDecreaseRedeemOrder(_msg)) {
+            return hex"6501";
+        } else if (Messages.isCollectInvest(_msg)) {
+            return hex"e4";
+        } else if (Messages.isCollectRedeem(_msg)) {
+            return hex"e4";
+        } else {
+            revert("AxelarXCMRelayer/unsupported-outgoing-message");
+        }
```

Comment:
@hieronx this fails because it doesn't know `Messages` 🙃  an alternative is that we have a storage `messag_call_id -> scale encoded length`, which also makes the contract more dynamic instead of requiring a re-deploy every time we change or add a new incoming message. wdyt?

Reply:
I don't understand why it's not possible to use the length of the `_msg` here?

Reply:
But for now, I would do what is easiest. I think a storage map for this works

Reply:
@hieronx because the length needs to be SCALE-encoded so that the call is decodable on the centrifuge chain side (Substrate chain expecting length in the SCALE format)

Reply:
So why do we have our own codec in the first place?

Reply:
The interface of `process_msg` expects an `BoundedVec<u8>` and calls `deserialize()` in the remaning bytes. Why is scale needed here?